components/formats-api/src/loci/formats/services/OMEXMLServiceImpl.java

```diff
@@ -554,6 +554,57 @@ public boolean validateOMEXML(String xml, boolean pixelsHack) {
         return false;
       }
     }
+
+    // remove any Modulo annotations for validation
+    // the "Label" attribute on Modulo conflicts with the "Label" shape
+    // and will cause validation errors
+
+    try {
+      OMEXMLMetadata omexml = createOMEXMLMetadata(xml);
+      OMEXMLMetadataRoot root = (OMEXMLMetadataRoot) omexml.getRoot();
+      for (int image=0; image<root.sizeOfImageList(); image++) {
+        Image img = root.getImage(image);
+
+        for (int i=0; i<img.sizeOfLinkedAnnotationList(); i++) {
+          Annotation annotation = img.getLinkedAnnotation(i);
+          if (!(annotation instanceof XMLAnnotation)) {
+            continue;
+          }
+
+          String annotationXML = ((XMLAnnotation) annotation).getValue();
+
+          Document annotationRoot = XMLTools.parseDOM(annotationXML);
+          NodeList nodes = annotationRoot.getElementsByTagName("ModuloAlongZ");
+
+          if (nodes.getLength() > 0) {
+            ((XMLAnnotation) annotation).setValue("");
+          }
+          nodes = annotationRoot.getElementsByTagName("ModuloAlongC");
+          if (nodes.getLength() > 0) {
+            ((XMLAnnotation) annotation).setValue("");
+          }
+          nodes = annotationRoot.getElementsByTagName("ModuloAlongT");
+          if (nodes.getLength() > 0) {
+            ((XMLAnnotation) annotation).setValue("");
+          }
+        }
+      }
+      omexml.setRoot(root);
+      xml = getOMEXML(omexml);
+    }
+    catch (ServiceException e) {
```

Comment:
just use multi-catch? (hmmm, can't find where we say which version of Java we require...)